docs/development/writingzeppelinspell.md

```diff
@@ -0,0 +1,213 @@
+---
+layout: page
+title: "Writing a new Spell(Experimental)"
+description: "Spell is a kind of interpreter that runs on browser not on backend. So, technically it's the frontend interpreter. "
+group: development
+---
+<!--
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+{% include JB/setup %}
+
+# Writing a new Spell (Experimental)
+
+<div id="toc"></div>
+
+## What is Apache Zeppelin Spell 
+
+Spell is a kind of interpreter that runs on browser not on backend. So, technically it's the frontend interpreter. 
+It can provide many benefits.
+
+- Spell is pluggable frontend interpreter. So it can be installed and removed easily using helium registry. 
+- Every spell is written in javascript. It means you can use existing javascript libraries whatever you want.
+- Spell runs on browser like display system (`%html`, `%table`). In other words, every spell can be used as display system as well.
+
+## How it works
+
+Helium Spell works like [Helium Visualization](./writingzeppelinvisualization.html).
+
+- Every helium packages are loaded from central (online) registry or local registry
+- You can see loaded packages in `/helium` page.
+- When you enable a spell, it's built from server and sent to client
+- Finally it will be loaded into browser.
+
+## How to use spell
+
+### 1. Enabling
+
+Find a spell what you want to use in `/helium` package and click `Enable` button.
+
+<img class="img-responsive" style="width:70%" src="../assets/themes/zeppelin/img/docs-img/writing_spell_registered.png" />
+
+### 2. Using
+
+Spell works like an interpreter. Thus, use the `MAGIC` to execute spell in a note. (you might need to refresh after enabling)
```

Comment:
this might be a bit hard to understand, how about say `for example, use %echo for example with the Echo Spell`